alns/accept/tests/test_record_to_record_travel.py

```diff
@@ -1,105 +1,84 @@
-    be accepted.
-    """
-    RecordToRecordTravel(1, 0, 1)
-
-
-def test_raises_start_smaller_than_end():
-    """
-    The initial threshold at the start should be bigger (or equal) to the end
-    threshold.
-    """
+@mark.parametrize(
+    "start, end, step, method",
+    [
+        (-1, 1, 1, "linear"),  # start threshold cannot be negative
+        (1, -1, 1, "linear"),  # nor can the end threshold
+        (1, 1, -1, "linear"),  # nor the updating step
+        (0, 1, 1, "linear"),  # start threshold cannot be < end threshold
+        (2, 1, 2, "exponential"),  # step cannot be > 1 with exponential
+        (1, 1, 1, "non-linear"),  # method must be 'linear' or 'exponential'
+    ],
+)
+def test_raises_invalid_parameters(start, end, step, method):
     with assert_raises(ValueError):
-        RecordToRecordTravel(0, 1, 1)
-
-    RecordToRecordTravel(1, 1, 1)  # should not raise for equality
-
-
-def test_does_not_raise():
-    """
-    This set of parameters should work correctly.
-    """
-    RecordToRecordTravel(10, 5, 2)
-
-
-def test_step():
-    """
-    Tests if the step parameter is correctly set.
-    """
-    for step in range(100):
-        assert_equal(RecordToRecordTravel(1, 1, step).step, step)
-
-
-def test_start_threshold():
-    """
-    Tests if the start_threshold parameter is correctly set.
-    """
-    for start in range(100):
-        assert_equal(RecordToRecordTravel(start, 0, 1).start_threshold, start)
-
-
-def test_end_threshold():
-    """
-    Tests if the end_threshold parameter is correctly set.
-    """
-    for end in range(100):
-        assert_equal(RecordToRecordTravel(100, end, 1).end_threshold, end)
+        RecordToRecordTravel(start, end, step, method)
+
+
+@mark.parametrize(
+    "start, end, step, method",
+    [
+        (0, 0, 1, "linear"),  # boundary start and end threshold
+        (0.5, 0.5, 1, "linear"),  # non-boundary equal start and end
+        (10, 0, 0, "linear"),  # boundary step
+        (1, 0.5, 1, "exponential"),  # boundary step exponential
+        (1000, 0.5, 0.1, "linear"),  # regular parameters
+    ],
+)
+def test_no_raise_valid_parameters(start, end, step, method):
+    RecordToRecordTravel(start, end, step, method)
+
+
+@mark.parametrize(
+    "start, end, step, method",
+    [
+        (1, 0, 1, "linear"),
+        (0.5, 0.5, 1, "linear"),
+        (10, 5, 0.001, "linear"),
+        (1, 0.5, 1, "exponential"),
+    ],
+)
+def test_properties(start, end, step, method):
+    """
+    Test if the properties are correctly set.
+    """
+    rrt = RecordToRecordTravel(start, end, step, method)
+    assert_equal(rrt.start_threshold, start)
+    assert_equal(rrt.end_threshold, end)
+    assert_equal(rrt.step, step)
+    assert_equal(rrt.method, method)
 
 
 def test_accepts_better():
     record_travel = RecordToRecordTravel(1, 0, 0.1)
-    assert_(record_travel(rnd.RandomState(), One(), Zero(), Zero()))
+    assert_(record_travel(None, One(), None, Zero()))
```

Comment:
Refactor: I'm not a big fan of using `None` here. The types do not allow this, and it is not much more convenient than just passing in the original values.

Reply:
The main motivation for using None was to make the tests more readable. I did this also for tests that are already on the main branch - want me to change that as well?

Reply:
Preferably yes!